objects/69ead2/contained/DUESETUP/script.lua

```diff
@@ -0,0 +1,644 @@
+            label = "☑\n\n\n",
+        })
+        createAllG3Buttons()
+    end
+    updateSave()
+end
+
+--------------------------------------
+-- impended card buttons and functions
+--------------------------------------
+
+function createCardButtons(card)
+    if not getButtonIndex(card,"energyDisplay") then -- stop multiple buttons appearing, did prevent another way before but that removed other functionality
+        card.createButton({ 
+            click_function = "energyDisplay",
+            function_owner = self,
+            label = "0/0",
+            position = {0.29,aboveCard,-0.55},
+            width = 700,
+            height = 550,
+            font_size = 430,
+            color = {0,0,0,buttonVisibility},
+            font_color = {1,1,1,100},
+            tooltip = "Select a card to impend",
+        })
+        
+        card.createButton({
+            click_function = "overridePlus",
+            function_owner = self,
+            label = "+",
+            position = {-0.2,-aboveCard,0.8},
+            rotation = {180,180,0},
+            width = 200,
+            height = 200,
+            font_size = 250,
+            color = {0,0,0,0.9},
+            font_color = {1,0,0,100},
+            tooltip = "Press this to override the value of impended energy.\nUseful for if time passes twice.",
+        })
+        
+        card.createButton({
+            click_function = "overrideMinus",
+            function_owner = self,
+            label = "-",
+            position = {0.2,-aboveCard,0.8},
+            rotation = {180,180,0},
+            width = 200,
+            height = 200,
+            font_size = 250,
+            color = {0,0,0,0.9},
+            font_color = {1,0,0,100},
+            tooltip = "Press this to override the value of impended energy.\nUseful for if time passes twice.",
+        })
+        
+        card.createButton({
+            click_function = "clearAll",
+            function_owner = self,
+            label = "FORCE RESET",
+            position = {0,-aboveCard,1.2},
+            rotation = {180,180,0},
+            width = 950,
+            height = 200,
+            font_size = 150,
+            color = {0,0,0,0.9},
+            font_color = {1,0,0,100},
+            tooltip = "Press this to delete the stored card data\nUse this for if impended card is forgotten\nThis cannot be undone!",
+        })
+    end
+end
+    
+function energyDisplay(card,_color,alt_click)
+    -- probably need tighter restriction that just turnSelected, since could be forgotten in fast phase. use Global.getVar("currentPhase")
+    local tbl = impendTable[card.guid]
+    local spiritColor = Global.call("getSpiritColor", {name = spiritName})
+    if tbl.turnSelected == Global.getVar("turn") then
+        if not alt_click then -- check if have energy and need energy
+            if Global.getVar("selectedColors")[spiritColor].counter.getValue() > 0 and tbl.energy[1] < tbl.maxEnergy then
```

Comment:
this should be `getTable` as well, though based on some of the other files touched in review, seems i missed this in the spirit/aspect setup code